### PR TITLE
Example for refactoring (part of) a legacy rule to Python

### PR DESCRIPTION
**Pull request by lwesterhof**

### CODE REVIEW

File: util/user.py [hunk 1]
```diff
@@ -57,11 +57,17 @@ def from_str(ctx, s):
...
+    :param ctx:  Combined type of a callback and rei struct
+    :param user: Given user
+
+    :returns: Boolean indicating if user exists
+    """
     if type(user) is str:
         user = from_str(ctx, user)
 
-    return genquery.Query(ctx, "USER_TYPE", "USER_NAME = '{}' AND USER_ZONE = '{}'".format(*user)).first() is not None
+    return genquery.Query(ctx, "USER_TYPE", "USER_NAME = '{}' AND USER_ZONE = '{}'".format(*user)).first() in ["rodsuser", "rodsadmin"]
```
Comment (FuHsinyu) [inline util/user.py:70]: I am curious why only ('rodsuser' or 'rodsadmin') are considered user, what about other user types such as rodsgroup?

File: groups.py [hunk 2]
```diff
@@ -705,6 +706,18 @@ def group_user_exists(ctx, group_name, username, include_readonly):
         return False
 
 
+@rule.make(inputs=[0], outputs=[1])
+def rule_user_exists(ctx, username):
+    """Rule wrapper to check if a user exists.
```
Comment (FuHsinyu) [inline groups.py:711]: It'd be nice to also add "('rodsuser' or 'rodsadmin')" for better clarification in the doc strings

File: util/user.py (hunk 1 above)
Comment (lwesterhof) [reply util/user.py:70]: In the context of the user utilities we only consider 'rodsuser' and 'rodsadmin' users. For 'rodsgroup' user types we have the group utilities.

**Review by FuHsinyu — COMMENTED** [review]
Look generally good. Tested with multiple users and all were success. See inline comments for a few clarification.

**Review by stsnel — APPROVED** [review]
LGTM as well :+1:

**Pull request merged**